doc/source/whatsnew/v1.3.0.rst

```diff
@@ -65,6 +65,41 @@ Notable bug fixes
 These are bug fixes that might have notable behavior changes.
 
 
+Preserve dtypes in  :meth:`~pandas.DataFrame.combine_first`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:meth:`~pandas.DataFrame.combine_first` will now preserve dtypes (:issue:`7509`)
+
+*pandas 1.2.x*
+
+.. code-block:: ipython
+
+   In [1]: df1 = pd.DataFrame({"A": [1, 2, 3], "B": [1, 2, 3]}, index=[0, 1, 2])
+   df2 = pd.DataFrame({"B": [4, 5, 6], "C": [1, 2, 3]}, index=[2, 3, 4])
+   combined = df1.combine_first(df2)
+   (combined, "---------------", combined.dtypes)
+   Out[2]:
+   (     A    B    C
+   0  1.0  1.0  NaN
+   1  2.0  2.0  NaN
+   2  3.0  3.0  1.0
+   3  NaN  5.0  2.0
+   4  NaN  6.0  3.0,
+   '---------------',
+   A    float64
+   B    float64
+   C    float64
+   dtype: object)
+
+*pandas 1.3.0*
+
+.. ipython:: python
+
+   df1 = pd.DataFrame({"A": [1, 2, 3], "B": [1, 2, 3]}, index=[0, 1, 2])
```

Comment:
you can do the construction in an ioythok block before both of these as it's the same

only show the combined (and dtypes) in the before / after